core/primitives/decode_stream.hpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef FILECOIN_CORE_PRIMITIVES_DECODE_STREAM_HPP
+#define FILECOIN_CORE_PRIMITIVES_DECODE_STREAM_HPP
+
+#include <string_view>
+
+#include "common/outcome.hpp"
+#include "common/hexutil.hpp"
+#include "common/textutil.hpp"
+
+namespace primitives {
+
+  /**
+   * Base class for representing data streams
+   */
+  template <typename T>
+  class DecodeStream {
```

Comment:
Maybe prohibit instantiation for signed types? It would solve a number of problems.

Reply:
primitives::DecodeStream is common representation, which can be used for any purposes, signed/unsigned logic exists only in child LEB128 implementation, where instantiation for signed types was prohibited by static_assert